File: vetix/plugins/binary_file.py
from vetix.plugin import Issue, Plugin, Severity
from vetix.utils.file_classifier import is_binary_file
from vetix.utils.utils import get_relative_path


class BinaryFileCheckPlugin(Plugin):
    """Detect suspicious binary files in SKILL directories."""

    name = "Binary File"

    def scan(self, skill_dir: str, file_path: str, content: str) -> list[Issue]:
        if not is_binary_file(file_path):
            return []

        relative_path = get_relative_path(file_path, skill_dir)
        return [Issue(
            name="Binary file",
            severity=Severity.HIGH,
            category="Obfuscation",
            description="Suspicious binary files were found in the SKILL directory.",
            file_path=relative_path,
            audit_required=False
        )]

File: internal/plugin/binary_file.go
package plugin

import "vetix/internal/pluginutils"

type BinaryFileCheckPlugin struct{}

func (BinaryFileCheckPlugin) Meta() Meta {
	return Meta{
		ID:          "binary_file",
		Name:        "Binary File",
		Description: "Flags binary files inside the SKILL directory, where payloads are invisible to text-based analysis.",
	}
}

func (BinaryFileCheckPlugin) Scan(skillDir, filePath, content string) []Issue {
	if !pluginutils.IsBinaryFile(filePath) {
		return nil
	}
	return []Issue{{
		Name:          "Binary file",
		Severity:      SeverityHigh,
		Category:      CatObfuscation,
		Description:   "Suspicious binary files were found in the SKILL directory.",
		FilePath:      relativePath(filePath, skillDir),
		AuditRequired: false,
	}}
}
